src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_battery_power_formula.py

```diff
@@ -65,39 +69,106 @@ def generate(
 
         component_graph = connection_manager.get().component_graph
 
-        battery_inverters = frozenset(
-            frozenset(
+        # all_connected_batteries has to be set because
+        # multiple inverters can be connected to the same battery
```

Comment:
It has a different meaning otherwise, which is confusing.
```suggestion
        # all_connected_batteries has to be a `set` because
        # multiple inverters can be connected to the same battery

Reply:
done